x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/swimlane/swimlane_connectors.tsx

```diff
@@ -0,0 +1,385 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { Fragment, useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import {
+  EuiCallOut,
+  EuiFieldText,
+  EuiFormRow,
+  EuiLink,
+  EuiSpacer,
+  EuiText,
+  EuiHorizontalRule,
+  EuiComboBox,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import * as i18n from './translations';
+import { ActionConnectorFieldsProps } from '../../../../types';
+import { SwimlaneActionConnector, SwimlaneConfig, SwimlaneFieldMap } from './types';
+import { useKibana } from '../../../../common/lib/kibana';
+import { getApplication } from './api';
+
+const SwimlaneActionConnectorFields: React.FunctionComponent<
+  ActionConnectorFieldsProps<SwimlaneActionConnector>
+> = ({ consumer, errors, action, editActionConfig, editActionSecrets, readOnly }) => {
+  const {
+    http,
+    notifications: { toasts },
+  } = useKibana().services;
+
+  const { docLinks } = useKibana().services;
+  const defaultNames: SwimlaneFieldMap[] = useMemo(
```

Comment:
I am wondering if we can just have a defaultMappings right way like 
```
const defaultMappings = {
 alertSourceKeyName: { fieldKey: 'alert-source', fieldId: ''},
 ...
};
```
it will simplify line 58 and we do not need to use `useMemo`